core/src/main/java/smile/classification/OneVersusRest.java

```diff
@@ -155,13 +225,17 @@ public int predict(T x) {
 
     @Override
     public int predict(T x, double[] posteriori) {
-        if (platts == null) {
-            throw new UnsupportedOperationException("Platt scaling is not available");
-        }
-
-        for (int i = 0; i < k; i++) {
-            posteriori[i] = platts[i].scale(classifiers[i].f(x));
-        }
+            if (classifiers[0] instanceof SoftClassifier) {
+                for (int i = 0; i < k; i++) {
+                    double[] proba = new double[2];
+                    ((SoftClassifier<T>) classifiers[i]).predict(x, proba);
+                    posteriori[i] = proba[1];
```

Comment:
This is wrong. posterior probability is not defined this. Your sum is eve not 1.